main.cpp

```diff
@@ -0,0 +1,58 @@
+#include <iostream>
+#include "queue.h"
+using namespace std;
+
+int main()
+{
+	/*
+    char c;
+    std::cout << "Type of queue (a - array, l - list): ";
+    std::cin >> c;
+    */
+    //BasicQueue<int> * queue = NULL;
+	/*
+    if (c == 'a')
+        queue = new ArrayQueue<int>;
+    else if (c == 'l')
+        queue = new ListQueue<int>;
+    else {
+        std::cout << "Bad type, exit" << std::endl;
+        return 1;
+    }
+	*/
+	int x = 7, y = 8;
+	BasicQueue<int> * queue = NULL;
+
+	/*
+	ListQueue<int> * queue_l = new ListQueue<int>;
+	queue_l->enqueue(x);
+	queue_l->enqueue(y);
+	queue = queue_l;
+	cout << queue_l->getLength() << endl;
+	cout << queue_l->dequeue() << endl;
+	cout << queue_l->getLength() << endl;
+	cout << queue_l->dequeue() << endl;
+	cout << queue_l->getLength() << endl;
+	*/
+	try {
+		ArrayQueue<int> * queue_a = new ArrayQueue<int>;

```

Comment:
Where are tests with ListQueue? Why you don't use polymorphism?
So many questions...